script.js

```diff
@@ -1,4 +1,213 @@
+  console.log("secondToGoFirst:", secondToGoFirst);
+  if (input == 1) {
+    playerNumberCombination2 = oneToGoFirst;
+  } else if (input == 2) {
+    playerNumberCombination2 = secondToGoFirst;
+  } else {
+    `please insert '1' to go with the first number first and '2' to go the the second number first.`;
+  }
+  console.log("playerNumberCombination2:", playerNumberCombination2);
+  return playerNumberCombination2;
+};
+
+//compare two players combination numbers, higher numbers win
+var compareNumbers = function () {
+  var compareNumberResult;
+  if (playerNumberCombination1 > playerNumberCombination2) {
+    compareNumberResult = `Player ONE final number: ${playerNumberCombination1} is larger than Player TWO final number: ${playerNumberCombination2}. Player ONE wins!`;
+  } else if (playerNumberCombination2 > playerNumberCombination1) {
+    compareNumberResult = `Player TWO final number: ${playerNumberCombination2} is larger than Player ONE final number: ${playerNumberCombination1}. Player TWO wins!`;
+  } else if (playerNumberCombination1 == playerNumberCombination2) {
+    compareNumberResult = `Player ONE final number is equal to Player TWO final number: ${playerNumberCombination1}. It's a draw!`;
+  }
+  return compareNumberResult;
+};
+
+//main function
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  // if two dice rolls result hasnt shown, output two dice rolls result
+  if (gameMode == "startGame") {
+    myOutputValue = getTwoDiceRollResult1();
+    gameMode = "pickNumberToGoFirst";
+  }
+  // if two dice rolls has generated, pick which number to go first
+  else if (gameMode == "pickNumberToGoFirst") {
+    if (input == 1 || input == 2) {myOutputValue = `Your number is ${pickNumberToGoFirst1(input)}.
+        Press <b>'Submit'</b> to generate Player TWO's number.`;
+      gameMode = "getSecondPlayerTwoDiceRoll";
+    }
+    else {gameMode = "pickNumberToGoFirst"};
+  }
+  // if first player number is picked, generate second player two dice results
+  else if (gameMode == "getSecondPlayerTwoDiceRoll") {
+    myOutputValue = getTwoDiceRollResult2();
+    gameMode = "pickSecondPlayerNumberToGoFirst";
+  }
+  // if second player two dice results has generated, second player pick which number to go first
+  else if (gameMode == "pickSecondPlayerNumberToGoFirst") {
+    if (input == 1 || input == 2) {myOutputValue = `Your number is ${pickNumberToGoFirst2(input)}.<br>
+      Press <b>'Submit'</b> to compare result.`;
+      gameMode = "compareNumbers";
+    }
+    else {gameMode = "pickSecondPlayerNumberToGoFirst"};
+  }
+  // if both players final number is set, compare both numbers
+  else if (gameMode == "compareNumbers") {
+    myOutputValue = `${compareNumbers()} <br><br>
+      Past Numbers of Player ONE is ${addPastNumbers1()}.<br>
+      Past Numbers of Player TWO is ${addPastNumbers2()}.<br><br>
+      ${showLeaderBoard()} <br><br>
+      Click "Submit" to continue the game.`;
```

Comment:
hmm, i think it might be clearer/ easier for the reader if you initialised variables and store the return value of these fucntions you're executing into the variables